java-checks/src/main/java/org/sonar/java/checks/SelfAssignementCheck.java

```diff
@@ -54,28 +69,77 @@ public void visitNode(Tree tree) {
     }
     AssignmentExpressionTree node = (AssignmentExpressionTree) tree;
     if (SyntacticEquivalence.areEquivalent(node.expression(), node.variable())) {
-      reportIssue(reportTree(node), ISSUE_MESSAGE);
+      QuickFixHelper.newIssue(context)
+        .forRule(this)
+        .onTree(reportTree(node))
+        .withMessage(ISSUE_MESSAGE)
+        .withQuickFix(() -> getQuickFix(node))
+        .report();
       updateWarnings(node);
     }
   }
 
+  private static JavaQuickFix getQuickFix(AssignmentExpressionTree tree) {
+    ClassTree classParent = (ClassTree) ExpressionUtils.getParentOfType(tree, Tree.Kind.CLASS);
+    MethodTree methodParent = (MethodTree) ExpressionUtils.getParentOfType(tree, Tree.Kind.METHOD, Tree.Kind.CONSTRUCTOR);
+    String name = getName(tree.variable());
+
+    if (methodParent != null && classParent != null) {
+      boolean isParameter = methodParent.parameters().stream()
+        .map(p -> p.simpleName().name())
+        .anyMatch(p -> p.equals(name));
+
+      if (isParameter) {
+        List<String> memberNames = classParent.members().stream()
+          .filter(m -> m.is(Tree.Kind.VARIABLE))
+          .map(VariableTree.class::cast)
+          .map(m -> m.simpleName().name())
+          .collect(Collectors.toList());
+
+        if (memberNames.contains(name)) {
+          return JavaQuickFix.newQuickFix("Correct this useless self-assignment")
```

Comment:
I think the quick fix message is a bit contradictory: we advise correcting the statement but also this is a useless statement.
How about changing the message?
```suggestion
          return JavaQuickFix.newQuickFix("Disambiguate this self-assignment")
```